ansible/roles/sysprep/tasks/redhat.yml

```diff
@@ -41,8 +41,36 @@
 - name: Remove yum package lists
   command: "{{ sysusr_prefix }}/bin/yum -y clean all"
 
-- name: Reset network interface IDs
-  shell: sed -i '/^\(HWADDR\|UUID\)=/d' /etc/sysconfig/network-scripts/ifcfg-*
+- name: find network profiles at /etc/sysconfig/network-scripts/
+  find:
+    paths: "/etc/sysconfig/network-scripts/"
+    patterns: "ifcfg-*"
+    file_type: file
+  register: network_profiles_sysconfig
+  changed_when: false
+
+- name: Reset network interface IDs for RHEL at /etc/sysconfig/network-scripts/
+  shell: sed -i '/^\(HWADDR\|UUID\)=/d' "{{item}}"
+  with_items: "{{  network_profiles_sysconfig.files | map(attribute='path') | list }}"
+
+# NetworkManager stores new network profiles in keyfile format in the
+# /etc/NetworkManager/system-connections/ directory.
+
+# Previously, NetworkManager stored network profiles in ifcfg format
+# in this directory (/etc/sysconfig/network-scripts/). However, the ifcfg
+# format is deprecated. By default, NetworkManager no longer creates
+# new profiles in this format.
+- name: find network profiles at /etc/NetworkManager/system-connections
```

Comment:
Curious how did you know that this is needed?

Reply:
I found this during testing offline mode. We do not update/install common rpms if it is already installed on the system. 
The network manager version that comes with RockyLinux DVD ISO uses new format and location for network profiles. New version of NetworkManager drops a readme(deprecation notice) at existing location `/etc/sysconfig/network-scripts/` explaining about new location and format.

It is possible that the rockylinux base AMI and azure uses a NetworkManager version that still uses `/etc/sysconfig/network-scripts/` for network profiles evident by previously passing aws and azure e2e tests. 
So I created these idempotent tasks to handle both case without guessing networkManger version or OS version.